### DISCUSSION TITLE
Zero-hardware steno machine

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1367] [Ideas]
Just an idea... Expect it to take a few years of research. (I don't have that much free time.)

It's possible to

* Make a shape of the steno keyboard from plastic or something
* Set up some camera to capture the image of the fingers
* Use some program on the computer to determine if the fingers touch the "keys"
* Feed the output to Plover with a machine plugin.

Of course the third step is hard.

**Comment by tfer** [comment]
I have some plans along this line of thinking.  Use frustrated total reflection, (FTR).  Take a sheet of acrylic and edge illuminate with infrared diodes, your fingers will break the light out plastic and be visible to the camera.  I'd probably use two sheets, one for the fingers, and, at a lower level, one for the thumbs.  Note the camera is underneath.

For hardware, I'm thinking something in the Nvidia Jetson line.
